modules/services/espanso.nix

```
@@ -0,0 +1,78 @@
+{ pkgs, config, lib, ... }:
+let
+  inherit (lib)
+    mkOption mkEnableOption mkIf maintainers literalExpression types platforms;
+    inherit (lib.hm.assertions) assertPlatform;
+  cfg = config.services.espanso;
+  yaml = pkgs.formats.yaml { };
+in {
+  meta.maintainers = with maintainers; [ lucasew ];
+
+  options = {
+    services.espanso = {
+      enable = mkEnableOption "Espanso: cross platform text expander in Rust";
+      package = mkOption {
+        type = types.package;
+        description = "Which espanso package to use";
+        default = pkgs.espanso;
+        defaultText = literalExpression "pkgs.espanso";
+      };
+      config = mkOption {
+        type = yaml.type;
+        description = "Espanso configuration";
+        default = { matches = [ ]; };
+        example = literalExpression ''
+          {
+            matches = [
+              { # Simple text replacement
+                trigger = ":espanso";
+                replace = "Hi there!";
+              }
+              { # Dates
+                trigger = ":date";
+                replace = "{{mydate}}";
+                vars = [{
+                  name = "mydate";
+                  type = "date";
+                  params = { format = "%m/%d/%Y"; };
+                }];
+              }
+              { # Shell commands
+                trigger = ":shell";
+                replace = "{{output}}";
+                vars = [{
+                  name = "output";
+                  type = "shell";
+                  params = { cmd = "echo Hello from your shell"; };
+                }];
+              }
+            ];
+          }
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      (assertPlatform "services.espanso" pkgs platforms.linux)
+    ];
+    systemd.user.services = {
+      espanso = {
+        Unit = {
+          Description = "Espanso: cross platform text expander in Rust";
+        };
+        Service = {
+          Type = "exec";
+          ExecStart = "${cfg.package}/bin/espanso daemon";
+          Restart = "on-failure";
+        };
+        Install = { WantedBy = [ "default.target" ]; };
+      };
+    };
+
+    home.file.".config/espanso/default.yml".source =
+      yaml.generate "default.yml" cfg.config;
```

```suggestion
      yaml.generate "espanso-default.yml" cfg.config;
```
It makes it easier to grep in the nix store :laughing: